bobcat/bb-cumber/src/main/java/com/cognifide/qa/bb/cumber/BobcumberListener.java

```diff
@@ -0,0 +1,119 @@
+/*-
+ * #%L
+ * Bobcat
+ * %%
+ * Copyright (C) 2016 Cognifide Ltd.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.qa.bb.cumber;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang3.CharEncoding;
+import org.junit.runner.Description;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunListener;
+
+import com.google.common.collect.Sets;
+
+public class BobcumberListener extends RunListener {
+
+	private static final String FEATURE = "feature";
+
+	private static final String SCENARIO = "Scenario";
+
+	private static final Map<String, Set<String>> ADDED_FEATURES = new HashMap<>();
+
+	private static final String COLON = ":";
+
+	private Integer scenarioCounter = 0;
```

Comment:
I think that operations on this variable should be thread safe. Consider using AtomicInteger#incrementAndGet etc